### ISSUE TITLE
Extraneous spaces?

### ISSUE THREAD

**Issue by crides** [charmbracelet/glamour#40]
When using `glow` to render markdowns, it seems like there are a lot of useless spaces at the end of lines. You can see in this screenshot (run with `glow github.com/charmbracelet/glow | bat -A`):

![maim-2019-12-23--14-35-38](https://user-images.githubusercontent.com/15631555/71379616-91832c80-2591-11ea-9d86-ba097748168e.png)

**Comment by muesli** [comment]
Correct. This seems pointless when you're using the default style that doesn't use background colors, but it makes a difference with different styles. Consider this example:

![image](https://user-images.githubusercontent.com/146378/71379792-00ca4200-25cd-11ea-99b6-3b69ee8338d7.png)

I think we can try and trim padding spaces when there's no background color being applied.

Btw, if you're using `glow`/`glamour` to pipe its output to a pager, I'd also recommend explicitly setting a style, otherwise it'll detect not printing to a terminal and switch to the `notty` style.
